src/AspNet.Security.OpenId.Steam/SteamAuthenticationHandler.cs

```diff
@@ -96,12 +97,14 @@ public class SteamAuthenticationHandler : OpenIdAuthenticationHandler<SteamAuthe
                 throw new HttpRequestException("An error occurred while retrieving the user profile from Steam.");
             }
 
-            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+            var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
 
             // Try to extract the profile name of the authenticated user.
-            var profile = payload.Value<JObject>(SteamAuthenticationConstants.Parameters.Response)
-                                ?.Value<JArray>(SteamAuthenticationConstants.Parameters.Players)
-                            ?[0]?.Value<string>(SteamAuthenticationConstants.Parameters.Name);
+            var profile = payload.RootElement.GetProperty(SteamAuthenticationConstants.Parameters.Response)
+                                .GetProperty(SteamAuthenticationConstants.Parameters.Players)
+                                .EnumerateArray()
+                                .First()
```

Comment:
Should we harden this against the array being empty?

Reply:
@martincostello `JsonElement` is not nullable, if use `FirstOrDefault`, it cannot be access conditionally.

Reply:
But `First()` will throw if there are none?